common-types/src/main/scala/com/improving/app/common/test/ServiceTestContainerSpec.scala

```diff
@@ -0,0 +1,64 @@
+package com.improving.app.common.test
+
+import akka.actor.ActorSystem
+import com.dimafeng.testcontainers.{DockerComposeContainer, ExposedService}
+import com.dimafeng.testcontainers.scalatest.TestContainerForAll
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.time.{Millis, Minutes, Span}
+import org.testcontainers.containers.wait.strategy.Wait
+
+import java.io.File
+
+/**
+ * ServiceTestContainerSpec class is the typical pattern for testing services that are gRPC services in a
+ * Docker container. The service in question should be present in the docker-compose.yml.
+ *
+ * When running the test cases that extends this class, it is assumed that the user already has a docker container for
+ * persistence (Scylla-DB) and has already run the command 'sbt clean compile && sbt docker:stage && sbt docker:publishLocal'
+ * @param exposedPort The exposedPort is an Integer that should match the port exposed in docker-compose.yml
+ * @param serviceName The serviceName is a String that should match the port exposed in docker-compose.yml
+ */
+class ServiceTestContainerSpec(exposedPort: Integer, serviceName: String)
+  extends AnyFlatSpec
+    with TestContainerForAll
+    with Matchers
+    with ScalaFutures {
```

Comment:
You might want to add Retries in here and handle the fixture with retries like you did in #12